xds/internal/balancer/clusterresolver/configbuilder.go

```diff
@@ -190,7 +190,19 @@ func buildClusterImplConfigForEDS(g *nameGenerator, edsResp xdsresource.Endpoint
 		})
 	}
 
-	priorities := groupLocalitiesByPriority(edsResp.Localities)
+	var priorities [][]xdsresource.Locality
+	// Triggered by an EDS error before update, or empty localities list in a
+	// update. In either case want to create a priority, and send down empty
+	// address list, causing TF for that priority.
+	if len(edsResp.Localities) == 0 {
+		// "If any discovery mechanism instance experiences an error retrieving
+		// data, and it has not previously reported any results, it should
+		// report a result that is a single priority with no endpoints." - A37
+		priorities = make([][]xdsresource.Locality, 1)
+		priorities[0] = make([]xdsresource.Locality, 0)
```

Comment:
How about:

`priorities := [][]xdsresource.Locality{{}}`

Since the `priorities` var is defined as a slice of slice of `xdsresource.Locality`, the compiler can figure out the type of the elements in the slice (when defined statically) without them having to be mentioned explicitly. So, in the above statement `{}` indicates an zero-value `xdsresource.Locality`.

Reply:
Wouldn't the {} be a zero value of []xdsresource.Locality? Thus setting to nil rather than an empty slice? I don't think it matters though, I forget why I didn't want this to be nil I'll try switching it over.